### PR TITLE
Feat push metrics

### PR DESCRIPTION
**Pull request by juandspy**
# Description

Package with helping functions to use Prometheus Pushgateways.

## Type of change

- New feature (non-breaking change which adds functionality)

## Testing steps

Unit tests. Also moved this code from a private package where I used a real Pushgateway to test the code.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,23 +10,25 @@
...
-    * [`github.com/RedHatInsights/insights-operator-utils/responses`](#githubcomredhatinsightsinsights-operator-utilsresponses)
-    * [`github.com/RedHatInsights/insights-operator-utils/tests`](#githubcomredhatinsightsinsights-operator-utilstests)
-    * [`github.com/RedHatInsights/insights-operator-utils/types`](#githubcomredhatinsightsinsights-operator-utilstypes)
-* [How to use this library](#how-to-use-this-library)
-* [Configuration](#configuration)
-* [Contribution](#contribution)
-* [Testing](#testing)
-* [CI](#ci)
-* [Package manifest](#package-manifest)
+- [insights-operator Utils](#insights-operator-utils)
```
Comment (tisnik) [inline README.md:13]: it will be regenerated to older version

File: metrics/push/push.go [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+// UnregisterMetrics unregister all prometheus collectors safe in the collectors variable
+func UnregisterMetrics() error {
+	var errMsg = "cannot unregister metric"
+	log.Debug().
+		Int("total", len(collectors)).
+		Int("progress", 0).
+		Msg("Unregistering metrics")
+	for i, c := range collectors {
+		if ok := prometheus.Unregister(c); !ok {
+			log.Warn().Msg(errMsg)
```
Comment (tisnik) [inline metrics/push/push.go:38]: no error to be returned?

**Review by tisnik — COMMENTED** [review]
LGTM, just two nitpicks

File: README.md (hunk 1 above)
Comment (juandspy) [reply README.md:13]: WDYM?

File: metrics/push/push.go (hunk 2 above)
Comment (juandspy) [reply metrics/push/push.go:39]: thx, solved a bug fixing this :D

File: metrics/push/push_test.go [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+	})
+	t.Run("already registered metrics", func(t *testing.T) {
+		err := push.InitMetrics(testInitFunctions)
+		assert.NoError(t, err)
+	})
+}
+
+func TestPushMetrics(t *testing.T) {
+	err := push.InitMetrics(testInitFunctions)
+	require.NoError(t, err)
```
Comment (epapbak) [inline metrics/push/push_test.go:49]: use assert. Not worth importing require only for this line check

**Review by epapbak — APPROVED** [review]
lgtm =)

**Review by tisnik — APPROVED** [review]
LGTM

**Pull request merged**